### DISCUSSION TITLE
Passing arguments from the .desktop file to the binary

### DISCUSSION THREAD

**Discussion by Daeraxa** [AppImage/AppImageKit#1333] [Q&A]
Sorry if this is a stupid question but this is something that I keep coming back to and have been struggling to understand. We provide an AppImage as part of our releases of the Pulsar text editor but for ages now we have had to caveat that you may need to run the appimage with an extra argument of `--no-sandbox` to get the Electron dev tools to work.

As part of the build we do create a .desktop file that includes this argument:
```
[Desktop Entry]
Name=Pulsar
Exec=AppRun --no-sandbox %U
```

It is my understanding that the AppRun script is then meant to take those arguments and pass them onto the actual binary. It does have the right binary `BIN="$APPDIR/pulsar"` but if I run the appimage and look at the process I can see it mounting into `/tmp/` but running the binary without the arguments:
`
```
$ ps -ef | grep "pulsar"                                                                                                                               
dae       139126  139054  1 02:25 pts/3    00:00:01 /tmp/.mount_Linux.LPU6du/pulsar

```

Any ideas why the arguments aren't being passed on? There is probably some simple answer I'm missing here but we are doing this mostly automated from electron-builder so I've not done much of a deep dive into this before.

**Comment by probonopd** [comment]
I don't think we are passing in arguments from the desktop file for code simplicity. But you could have your custom AppRun file that would add things like `--no-sandbox`.
